tests/integration/suite/metadata/metadata.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/suite"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func init() {
+	suite.Register(new(Metadata))
+}
+
+// Metadata tests Dapr's response to metadata API requests.
+type Metadata struct{}
+
+func (*Metadata) Setup(*testing.T, context.Context) []framework.RunDaprdOption {
+	return nil
+}
+
+func (*Metadata) Run(t *testing.T, ctx context.Context, cmd *framework.Command) {
+	assert.Eventually(t, func() bool {
+		_, err := net.Dial("tcp", fmt.Sprintf("localhost:%d", cmd.InternalGRPCPort))
+		return err == nil
+	}, time.Second*5, 100*time.Millisecond)
+
+	reqURL := fmt.Sprintf("http://localhost:%d/v1.0/metadata", cmd.PublicPort)
+
+	assert.Eventually(t, func() bool {
```

Comment:
Why the `eventually` here? If you're already testing for the server to be up using the `net.Dial` above?

Reply:
True, I was following `apphealthz.go` test but looks like we can do without eventually, removed!